rosetta/docs/GPU_performance.md

```diff
@@ -0,0 +1,131 @@
+# Tips for High-Performance LLMs with JAX and XLA 
+
+This page documents the various flags in XLA and JAX to improve performance for LLMs on GPUs. The XLA flags are defined with their default values in [xla/debug_options_flags.cc](https://github.com/openxla/xla/blob/main/xla/debug_options_flags.cc)
+
+
+## Flags to manage memory used in JAX/XLA
+
+- XLA_PYTHON_CLIENT_MEM_FRACTION is a JAX flag that allocates a fraction of GPU memory for JAX/XLA.
+--  Ideally, should be 1, but in practice less because some memory is used by NVIDIA Libraries, and the JAX framework.
+--  We typically set it to 0.9 or 0.8. At 0.9, XLA gets 90% of GPU memory.
+
+- xla_gpu_memory_limit_slop_factor controls the memory used by XLA for determining its default heuristics for scheduling, and rematerialization. Default is recommended.
+
+
+## General CUDA/NCCL flags 
+
+### CUDA Flags
+
+The following flag restricts cuda queues to 1 and is useful when a strict ordering of operations is required to achieve best performance. This is recommended to achieve good performance with latency hiding optimizations with asynchronous collectives.
+- CUDA_DEVICE_MAX_CONNECTIONS=1
+  
+### NCCL flags 
+
+See [Environment Variables] (https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/env.html) for more details.
+- NCCL_LL128_BUFFSIZE: -2
+- NCCL_LL_BUFFSIZE: -2
```

Comment:
Those 2 aren't documented. What -2 mean? I suppose this is a special meaning, maybe add a quick sentense to explain what they do?

Reply:
https://github.com/NVIDIA/nccl/blob/master/src/init.cc#L552